src/targets.rs

```diff
@@ -96,6 +95,53 @@ impl Default for InitializationConfig {
     }
 }
 
+#[derive(Eq)]
+pub struct TargetTriple {
+    pub(crate) triple: LLVMString,
+}
+
+impl TargetTriple {
+    pub(crate) fn new(triple: LLVMString) -> TargetTriple {
+        TargetTriple {
+            triple,
+        }
+    }
+
+    pub fn create(triple: &str) -> TargetTriple {
+        let c_string = CString::new(triple).expect("Conversion to CString failed unexpectedly");
+
+        TargetTriple {
+            triple: LLVMString::create_from_c_str(c_string.as_c_str())
```

Comment:
You should just do `LLVMString::create_from_str(triple)` so as to not allocate with `CString`

Reply:
Hmm, when tests started failing I remembered the reason why I did this: if the user's &str is not NUL terminated then calling LLVMString::create_from_str will fail (and in general we don't expect it to be). This is a pretty fundamental Rust and C interop problem and I'm curious about your thoughts on it. We could avoid one of the allocations by cheating because we know the implementation of `LLVMCreateMessage` and `LLVMDisposeMessage`...

Reply:
Ack, you're right. I keep forgetting about this. lol. Let's leave it as it is then and I'll merge